internal/utils_test.go

```diff
@@ -93,11 +95,37 @@ func TestXSyncMapCounterMap(t *testing.T) {
 			go func() {
 				defer wg.Done()
 				cm.Inc("key")
+				cm.Inc("key2")
 			}()
 		}
 		wg.Wait()
 
-		assert.Equal(map[string]int64{"key": 10}, cm.GetAndReset())
+		assert.Equal(map[string]int64{"key": 10, "key2": 10}, cm.GetAndReset())
+	})
+
+	t.Run("concurrent with reset", func(t *testing.T) {
+		assert := assert.New(t)
+		cm := NewXSyncMapCounterMap()
+		wg := sync.WaitGroup{}
+		var val atomic.Int64
+		for range 10 {
+			wg.Add(2)
+			go func() {
+				defer wg.Done()
+				cm.Inc("key")
+			}()
+
+			go func() {
+				defer wg.Done()
+				v, ok := cm.GetAndReset()["key"]
+				if ok {
+					val.Add(v)
+				}
+
+			}()
+		}
+		wg.Wait()
```

Comment:
```suggestion
		wg.Wait()
		v, ok := cm.GetAndReset()["key"]
		if ok {
			val.Add(v)
		}
```
One additional `GetAndReset` might be needed.